apps/chat/src/components/Common/OptionsDialog.tsx

```diff
@@ -0,0 +1,76 @@
+import { useId } from 'react';
+
+import classNames from 'classnames';
+
+import { ModalState } from '@/src/types/modal';
+
+import Modal from '@/src/components/Common/Modal';
+
+const fakeCloseHandler = () => undefined;
+
+interface Props {
+  isOpen: boolean;
+  heading: string;
+  options: {
+    label: string;
+    dataQa: string;
+    className?: string;
+    onClick: () => void;
+  }[];
+  description?: string;
+  headingClassName?: string;
+  onClose?: () => void;
+}
+
+export const OptionsDialog = ({
+  heading,
+  headingClassName,
+  description,
+  isOpen,
+  onClose,
+  options,
+}: Props) => {
+  const descriptionId = useId();
+
+  return (
+    <Modal
+      portalId="theme-main"
+      state={isOpen ? ModalState.OPENED : ModalState.CLOSED}
+      onClose={onClose ?? fakeCloseHandler}
+      dataQa="options-dialog"
+      containerClassName="inline-block w-full min-w-[90%] px-3 py-4 md:p-6 text-center md:min-w-[300px] md:max-w-[500px]"
+      dismissProps={{ outsidePressEvent: 'mousedown', outsidePress: true }}
```

Comment:
it's new object each time. Let's move it in modal constants.
I suggest to create in constants file and use everywhere (see `modalDismissProps`):
- const MOUSE_OUTSIDE_PRESS_EVENT = { outsidePressEvent: 'mousedown' }
- const OUTSIDE_PRESS = { outsidePress: true }
- const OUTSIDE_PRESS_AND_MOUSE_EVENT = { ...MOUSE_OUTSIDE_PRESS_EVENT, ...OUTSIDE_PRESS  }